docs/upload_specifications.ADR.md

```diff
@@ -0,0 +1,170 @@
+
+## Resources
+
+* [The original strategy document for this approach](https://docs.google.com/document/d/1ocmZVBlTzAh_cdZaWGRwIbAuH-mcPRZFdhcwRhAfzxM/edit)
+  * Readable for everyone that has access to the KBase Google Docs folder
+  * The document contains short descriptions of future projects not included in this work,
+    such as generating template files on the fly and rich templates.
+
+## Front end changes
+
+The design introduces a new StS data type, `upload_specification`. The FE's current 
+behavior is to display any data types in the file dropdown, but ignore files for which
+the selected data type is unknown, a bug. The FE will be updated to ignore unknown
+data types returned from the StS, allowing for phased, non-lockstep upgrades. This work is not
+including is this project, but will be in a future FE project.
+
+## Upload specification input files
+
+Input file formats may be Excel, CSV, or TSV. An example CSV file structure for GFF/FASTA uploads
+is below:
+
+```
+Data type:, gff_metagenome
+fasta_file, gff_file, genome_name, source, release, genetic_code, generate_missing_genes
+FASTA File Path, GFF3 File Path, Metagenome Object Name, Source of metagenome, Release or Version of the Source Data, Genetic Code for protein translation, Spoof Genes for parentless CDS
+mygenome.fa, mygenome.gff3, mygenomeobject, , , 11, 0
+mygenome2.fa, mygenome2.gff3, mygenomeobject2, yermumspoo, 30456, 11, 1
+...
+```
+
+The file, by row, is:
+1. The data type, in this case `gff_metagenome`. This is a data type listed in the
+  [Mappings.py](https://github.com/kbase/staging_service/blob/master/staging_service/autodetect/Mappings.py)
+  file in the StS. The Narrative is expected to understand these types and map them to uploader
+  apps.
+2. The IDs of the app inputs from the `spec.json` file. 
+3. The corresponding human readable names of the app inputs from the `display.yaml` file.
+4. (and beyond) Import specifications. Each line corresponds to a single import.
+
+For Excel files, the first two rows may be hidden in any provided templates. Additionally,
+Excel files may contain multiple data types, one per tab.
+
+As part of this project we will deliver:
+1. CSV templates for each in scope app (e.g. the first 3 lines of the example file)
+2. An Excel template containing a tab for each in scope app
+3. A `README.md` file explaining how to use the templates.
+   * The `README.md` should include a link to rich upload specification documentation on the KBase
+     website once it is developed.
+   * Note: cover booleans which are not intuitive.
+
+These files will reside in the StS repo. As part of the front end effort, some means of
+delivering the templates and instructions to users will be developed.
+
+Currently, for all in scope apps, the inputs are all strings, numbers, or booleans. There are
+no unusual inputs such as grouped parameters or dynamic lookups. Including future upload apps
```

Comment:
Is the organism ID for GenBank genomes not a dynamic lookup?

Reply:
Not as yet.